pygac/reader.py

```diff
@@ -113,41 +116,83 @@ def __init__(self, interpolate_coords=True, adjust_clock_drift=True,
         self.lons = None
         self.times = None
         self.tle_lines = None
-        self.filename = None
         self._mask = None
 
+    @property
+    def filename(self):
+        """Get the property 'filename'."""
+        return self.__filename
```

Comment:
It is already initialized in the `__init__`. I added the filename to the argument list and therefore moved it upwards to the initialization of the other given arguments.
Since you are using the `Reader.read` method in satpy, I wanted to ask, if you keep a single reader instance and reuse it for may files, or if one reader instance represents one file.
In the former case, I would suggest to give the `Reader.read` the optional argument `fileobj` as seen in `file_opener` and keep setting the filename attribute in this method.

Reply:
Question: Should the filename attribute be initialized with the given filename, or would you prefer the variable `Reader.head['data_set_name']`?

Reply:
I believe using 'data_set_name' would be more robust, right ?